chain-impl-mockchain/src/account.rs

```diff
@@ -0,0 +1,181 @@
+use crate::value::*;
+use chain_crypto::{Ed25519Extended, PublicKey, SecretKey};
+use imhamt::{Hamt, InsertError, UpdateError};
+use std::collections::hash_map::DefaultHasher;
+
+type AccountAlg = Ed25519Extended;
+
+/// Possible errors during an account operation
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum LedgerError {
+    NonExistent,
+    AlreadyExists,
+    MismatchCounter,
+    NeedTotalWithdrawal,
+    NonZero,
+    ValueError(ValueError),
+}
+
+impl From<ValueError> for LedgerError {
+    fn from(e: ValueError) -> Self {
+        LedgerError::ValueError(e)
+    }
+}
+
+impl From<UpdateError<LedgerError>> for LedgerError {
+    fn from(e: UpdateError<LedgerError>) -> Self {
+        match e {
+            UpdateError::KeyNotFound => LedgerError::NonExistent,
+            UpdateError::ValueCallbackError(v) => v,
+        }
+    }
+}
+
+impl From<InsertError> for LedgerError {
+    fn from(e: InsertError) -> Self {
+        match e {
+            InsertError::EntryExists => LedgerError::AlreadyExists,
+        }
+    }
+}
+
+/// Account Identifier (also used as Public Key)
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Identifier(PublicKey<AccountAlg>);
+
+/// Account Secret Key
+pub struct Secret(SecretKey<AccountAlg>);
+
+#[derive(Clone)]
+pub struct State {
```

Comment:
`State` is a bit vague, maybe `AccountState` would be better (or even just `Account`)?

Reply:
it was `AccountState` originally, but I stripped the `Account` prefix in favor of using the module system